pkg/node/manager/manager.go

```diff
@@ -362,16 +362,6 @@ func (m *manager) backgroundSync(ctx context.Context) error {
 // legacyNodeIpBehavior returns true if the agent is still running in legacy
 // mode regarding node IPs
 func (m *manager) legacyNodeIpBehavior() bool {
-	// Cilium < 1.7 only exposed the Cilium internalIP to the ipcache
-	// unless encryption was enabled. This meant that for the majority of
-	// node IPs, CIDR policy rules would apply. With the introduction of
-	// remote-node identities, all node IPs were suddenly added to the
-	// ipcache. This resulted in a behavioral change. New deployments will
-	// provide this behavior out of the gate, existing deployments will
-	// have to opt into this by enabling remote-node identities.
-	if m.conf.RemoteNodeIdentitiesEnabled() {
-		return false
```

Comment:
This looks incorrect. `m.conf.RemoteNodeIdentitiesEnabled()` is true by default. This means that this function here basically should always return false now. However, as is, it always returns true.

Let's remove fully remove `legacyNodeIpBehavior`, as we can now assume that is is always false.

Reply:
Done